File: 0496-next-greater-element-i/0496-next-greater-element-i.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    const ans = [];
    for (let i = 0; i < nums1.length; i++) {
        const target = nums1[i];
        const index = nums2.indexOf(target);
        let nextGreater = -1;
        for (let j = index + 1; j < nums2.length; j++) {
            if (nums2[j] > target) {
                nextGreater = nums2[j];
                break;
            }
        }
        ans.push(nextGreater);
    }
    return ans;
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        stack = []
        next_greater = {}
        result = []

        for num in reversed(nums2):

            while stack and stack[-1] <= num:
                stack.pop()
            
            next_greater[num] = stack[-1] if stack else -1
            stack.append(num)

        for num in nums1:

            result.append(next_greater[num])
        return result